File: music-lessons-2-java/app/src/main/java/com/mstoyanov/musiclessons/model/PhoneNumberTypeConverter.java
package com.mstoyanov.musiclessons.model;

import android.arch.persistence.room.TypeConverter;

public class PhoneNumberTypeConverter {

    @TypeConverter
    public PhoneNumberType toPhoneNumberType(String phoneNumberType) {
        switch (phoneNumberType) {
            case "Home":
                return PhoneNumberType.HOME;
            case "Cell":
                return PhoneNumberType.CELL;
            case "Work":
                return PhoneNumberType.WORK;
            case "Other":
                return PhoneNumberType.OTHER;
            default:
                return null;
        }
    }

    @TypeConverter
    public String toString(PhoneNumberType phoneNumberType) {
        return phoneNumberType.getDisplayValue();
    }
}

File: app/src/main/kotlin/com/mstoyanov/musiclessons/model/PhoneNumberTypeConverter.kt
package com.mstoyanov.musiclessons.model

import androidx.room.TypeConverter

class PhoneNumberTypeConverter {
    // do not delete this method, IntelliJ doesn't see it being used, but it is:
    @TypeConverter
    fun toPhoneNumberType(value: String): PhoneNumberType? {
        return when (value) {
            "Home" -> PhoneNumberType.HOME
            "Cell" -> PhoneNumberType.CELL
            "Work" -> PhoneNumberType.WORK
            "Other" -> PhoneNumberType.OTHER
            else -> null
        }
    }

    @TypeConverter
    fun toString(phoneNumberType: PhoneNumberType): String {
        return phoneNumberType.displayValue()
    }
}
